pyiron_contrib/tinybase/creator.py

```diff
@@ -0,0 +1,255 @@
+                    exists, delete it first
+                delete_existing_job (bool): if a job if this name and type
+                    exists and its status is aborted, delete it first
+            """
+            try:
+                job = self._project.load(name)
+                if delete_existing_job or (
+                    delete_aborted_job and job.status == "aborted"
+                ):
+                    job.remove()
+                else:
+                    if not isinstance(job.task, task):
+                        raise ValueError(
+                            f"Job with given name already exists, but is of different type!"
+                        )
+                    return job
+            except JobNotFoundError:
+                pass
+            return pyiron_contrib.tinybase.job.TinyJob(task(), self._project, name)
+
+        return create
+
+    def register(self, task: Union["AbstractTask", str], name: str):
+        """
+        Register a new task.
+
+        Args:
+            task (AbstractTask): new task to register
+
+        Returns:
+            AbstractTask: the newly registered task
+        """
+        if name is None:
+            name = type(task).__name__
+        if name in self._jobs:
+            if task != self._jobs[name]:
+                raise ValueError(
+                    "Refusing to register task: different task of the same name already exists!"
+                )
+            else:
+                return self._jobs[name]
+        self._jobs[name] = task
+        return task
+
+
+class TaskCreator(Creator):
+    def __init__(self, project, config):
+        self._tasks = config.copy()
+
+    def __dir__(self):
+        return tuple(self._tasks.keys())
+
+    def __getattr__(self, name):
+        task = self._tasks[name]
+        if isinstance(task, str):
+            task = self._tasks[name] = load_class(task)
+        return task
+
+    def register(self, task: Union["AbstractTask", str], name: str):
+        """
+        Register a new task.
+
+        Args:
+            task (AbstractTask): new task to register
+
+        Returns:
+            AbstractTask: the newly registered task
+        """
+        if name is None:
+            name = type(task).__name__
+        if name in self._jobs:
+            if task != self._jobs[name]:
+                raise ValueError(
+                    "Refusing to register task: different task of the same name already exists!"
+                )
+            else:
+                return self._jobs[name]
+        self._jobs[name] = task
+        return task
+
+
+class StructureCreator(Creator):
+    def __init__(self, project, config):
+        pass
+
+    @wraps(ase.build.bulk)
+    def bulk(self, *args, **kwargs):
+        return ase_to_pyiron(ase.build.bulk(*args, **kwargs))
+
+    @wraps(Atoms)
+    def atoms(self, *args, **kwargs):
+        return Atoms(*args, **kwargs)
```

Comment:
What is the advantage to this over simply `atoms = Atoms`? I see that the defined method prevents overwriting the attribute, but I'm not so worried about that and otherwise I don't see any differences in functionality or documentation access.

For `bulk` I see it right away, because you really wrap it in `ase_to_pyiron` first (maybe there should be an addition to `bulk.__doc__` indicating that the return type changes), but here I'm just curious if I'm missing some helpful behaviour

Reply:
I didn't actually consider it.  It would have to be a method though, I guess, so 

```python
atoms = staticmethod(Atoms)
```

would work I suppose, but look a bit wonky to me.

Reply:
Is there a reason it needs to be a method? Over in workflows [I just slapped a whole class as an attribute on another class like this](https://github.com/pyiron/pyiron_contrib/blob/9378cc9e362f17fb91ecb9343435175a0f986386/pyiron_contrib/workflow/has_nodes.py#L146) and so far haven't run into any side effects.

Reply:
Fair I hadn't considered that.